src/backend/distributed/transaction/citus_dist_stat_activity.c

```diff
@@ -487,55 +489,11 @@ ParseCitusDistStat(PGresult *result, int64 rowIndex)
 {
 	CitusDistStat *citusDistStat = (CitusDistStat *) palloc0(sizeof(CitusDistStat));
 	int initiator_node_identifier = 0;
-	WorkerNode *initiatorWorkerNode = NULL;
 
-	/*
-	 * Replace initiator_node_identifier with initiator_node_hostname
-	 * and initiator_node_port given that those are a lot more useful.
-	 *
-	 * The rules are following:
-	 *    - If initiator_node_identifier belongs to a worker, simply get it
-	 *      from the metadata
-	 *   - If the initiator_node_identifier belongs to the coordinator and
-	 *     we're executing the function on the coordinator, get the localhost
-	 *     and port
-	 *   - If the initiator_node_identifier belongs to the coordinator and
-	 *     we're executing the function on a worker node, manually mark it
-	 *     as "coordinator_host" given that we cannot know the host and port
-	 *   - If the initiator_node_identifier doesn't equal to zero, we know that
-	 *     it is a worker query initiated outside of a distributed
-	 *     transaction. However, we cannot know which node has initiated
-	 *     the worker query.
-	 */
+
 	initiator_node_identifier =
 		PQgetisnull(result, rowIndex, 0) ? -1 : ParseIntField(result, rowIndex, 0);
-	if (initiator_node_identifier > 0)
-	{
-		bool nodeExists = false;
-
-		initiatorWorkerNode = PrimaryNodeForGroup(initiator_node_identifier, &nodeExists);
-
-		/* a query should run on an existing node */
-		Assert(nodeExists);
-		citusDistStat->master_query_host_name =
-			cstring_to_text(initiatorWorkerNode->workerName);
-		citusDistStat->master_query_host_port = initiatorWorkerNode->workerPort;
-	}
-	else if (initiator_node_identifier == 0 && IsCoordinator())
-	{
-		citusDistStat->master_query_host_name = cstring_to_text(coordinator_host_name);
-		citusDistStat->master_query_host_port = PostPortNumber;
-	}
-	else if (initiator_node_identifier == 0)
-	{
-		citusDistStat->master_query_host_name = cstring_to_text(coordinator_host_name);
-		citusDistStat->master_query_host_port = 0;
-	}
-	else
-	{
-		citusDistStat->master_query_host_name = NULL;
-		citusDistStat->master_query_host_port = 0;
-	}
+	ReplaceInitiatorNodeIdentifier(initiator_node_identifier, citusDistStat);
```

Comment:
Again super minor convention note, feel free to ignore: We generally try to have a single empty line between code-blocks that are logically different than others.

For example, when you use  `ReplaceInitiatorNodeIdentifier ` is feels a little bit different than `PQgetisnull ` / `ParseIntFieldFromHeapTuple `, so might be good to have a single line for readability purpose.

Try it out, if that doesn't feel similar for you, skip doing it.